cybox/common/vocabs.py

```diff
@@ -99,43 +157,1368 @@ def to_dict(self):
         return vocab_dict
 
     @classmethod
-    def from_obj(cls, vocab_obj):
+    def from_obj(cls, vocab_obj, return_obj=None):
         if not vocab_obj:
             return None
 
-        vocab_str = cls()
+        if not return_obj:
+            klass = VocabString.lookup_class(vocab_obj.xsi_type)
+            return_obj = klass()
+
         # xsi_type should be set automatically by the class's constructor.
 
-        vocab_str.vocab_name = vocab_obj.vocab_name
-        vocab_str.vocab_reference = vocab_obj.vocab_reference
-        vocab_str.xsi_type = vocab_obj.xsi_type
+        return_obj.vocab_name = vocab_obj.vocab_name
+        return_obj.vocab_reference = vocab_obj.vocab_reference
+        return_obj.xsi_type = vocab_obj.xsi_type
 
-        PatternFieldGroup.from_obj(vocab_obj, vocab_str)
+        PatternFieldGroup.from_obj(vocab_obj, return_obj)
 
         # We need to check for a non-default delimiter before trying to parse
         # the value.
-        vocab_str.value = denormalize_from_xml(vocab_obj.valueOf_,
-                                               vocab_str.delimiter)
+        return_obj.value = denormalize_from_xml(
+            value=vocab_obj.valueOf_,
+            delimiter=return_obj.delimiter
+        )
 
-        return vocab_str
+        return return_obj
 
     @classmethod
-    def from_dict(cls, vocab_dict):
+    def from_dict(cls, vocab_dict, return_obj=None):
         if not vocab_dict:
             return None
 
-        vocab_str = cls()
-        # xsi_type should be set automatically by the class's constructor.
+        if not return_obj:
+            if isinstance(vocab_dict, dict):
+                klass = VocabString.lookup_class(vocab_dict.get('xsi:type'))
+                return_obj = klass()
+            else:
+                return_obj = cls()
 
         # In case this is a "plain" string, just set it.
         if not isinstance(vocab_dict, dict):
-            vocab_str.value = vocab_dict
+            return_obj.value = vocab_dict
         else:
-            vocab_str.xsi_type = vocab_dict.get('xsi:type', cls._XSI_TYPE)
-            vocab_str.value = vocab_dict.get('value')
-            vocab_str.vocab_name = vocab_dict.get('vocab_name')
-            vocab_str.vocab_reference = vocab_dict.get('vocab_reference')
+            return_obj.xsi_type = vocab_dict.get('xsi:type', cls._XSI_TYPE)
+            return_obj.value = vocab_dict.get('value')
+            return_obj.vocab_name = vocab_dict.get('vocab_name')
+            return_obj.vocab_reference = vocab_dict.get('vocab_reference')
+
+            PatternFieldGroup.from_dict(vocab_dict, return_obj)
+
+        return return_obj
+
+
+class EventType(VocabString):
+    _namespace = 'http://cybox.mitre.org/default_vocabularies-2'
+    _XSI_TYPE = 'cyboxVocabs:EventTypeVocab-1.0.1'
+    _VOCAB_VERSION = '1.0.1'
+    _ALLOWED_VALUES = (
+        'File Ops (CRUD)',
```

Comment:
I would love for these to be sorted alphabetically, just for ease of searching in the source code (and possibly in the docs). I know that doesn't match the way they're defined in the schemas, and I don't care :wink: